File: app/src/main/java/com/example/map/domain/repository/LocalSearchRepository.java
package com.example.map.domain.repository;

import com.example.map.data.remote.model.LocalSearchResult;
import com.example.map.domain.request.SearchByAddressRequest;
import com.example.map.domain.request.SearchByCategoryRequest;
import com.example.map.domain.request.SearchByKeywordRequest;

import java.util.function.Consumer;

public interface LocalSearchRepository {
    void searchByAddress(SearchByAddressRequest request, Consumer<LocalSearchResult> onSuccess, Consumer<Throwable> onFailure);
    void searchByCategory(SearchByCategoryRequest request, Consumer<LocalSearchResult> onSuccess, Consumer<Throwable> onFailure);
    void searchByKeyword(SearchByKeywordRequest request, Consumer<LocalSearchResult> onSuccess, Consumer<Throwable> onFailure);
}

File: app/src/main/java/com/example/map/domain/repository/LocalSearchRepository.kt
package com.example.map.domain.repository

import com.example.map.data.remote.model.LocalSearchResult
import com.example.map.domain.request.SearchByAddressRequest
import com.example.map.domain.request.SearchByCategoryRequest
import com.example.map.domain.request.SearchByKeywordRequest

interface LocalSearchRepository {
    suspend fun searchByAddress(request: SearchByAddressRequest): Result<LocalSearchResult>

    suspend fun searchByCategory(request: SearchByCategoryRequest): Result<LocalSearchResult>

    suspend fun searchByKeyword(request: SearchByKeywordRequest): Result<LocalSearchResult>
}
